### ISSUE TITLE
NOAA EMC Land Data to IODA Conversion Software Support and Enhancement

### ISSUE THREAD

**Issue by emilyhcliu** [JCSDA-internal/ioda-converters#434]
## Description

This task is to support the community land Data Assimilation (Land DA) activities by developing ioda-converters for snow products (e.g., IMS snow cover fraction, AFWA snow depth, GHCN snow depth, GTS snow depth), surface soil moisture (e.g., ASCAT Level 2 NRT product for both 12.5 and 25 km resolution, SMAP Level 2 NRT product, and SMOS Level 2 NRT product), and the other surface variables such as 2m air temperature, dew point, and air pressure from NCEP bufr files.  These converters can convert native files with various formats (e.g., NETCDF, HP5, GRIB1/GRIB2) to ioda standard NETCDF files which can be directly used in JEDI. The first step is to initiate and develop codes with old ioda standard format to establish ioda converters for land. The second step is to enhance these converter codes to meet ioda-engine new standard format. Based on the users' requirement, we can process short-term and/or long-term land observation data.  These new converters and capability need to be added for the variables and/or products below:

- [ ]  AFWA Snow Depth
- [ ]  IMS Snow Cover Fraction
- [ ]  GHCN Snow Depth
- [ ]  ASCAT Soil Moisture Normalized
- [ ]  SMAP Soil Moisture
- [ ]  SMOS Soil Moisture
- [ ]  2m Air Temperature
- [ ]  2m Dew Point
- [ ]  Land Surface Pressure
- [ ]  GTS Snow Depth   

## **Documentation**

- [Land Task Table](https://drive.google.com/file/d/1k0RjGURzI0tWBFp-BFvtgfI3QU2VSDv4/view?usp=sharing)

## **Resources**

- AFWA Snow Depth
- [IMS Snow Cover](https://www.natice.noaa.gov/ims/)
- [GHCN Snow Depth](https://www.ncdc.noaa.gov/ghcnd-data-access)
- [ASCAT L2 NRT Soil Moisture Normalized](https://navigator.eumetsat.int/product/EO:EUM:DAT:METOP:SOMO25)
- [SMAP L2 NRT Soil Moisture](https://smap.jpl.nasa.gov/)
- [SMAP L2 Soil Moisture ](https://nsidc.org/data/smap/smap-data.html)
- [SMOS L2 NRT Soil Moisture](https://smos-diss.eo.esa.int/oads/access/)
- [NCEP BUFR/PREPBUFR Land Surface Products](https://www.nco.ncep.noaa.gov/sib/decoders/BUFRLIB/toc/prepbufr/)

## **References**

- Dharssi, I., K. J. Bovis, B. Macpherson, and C. P. Jones, 2011: Operational assimilation of ASCAT surface soil wetness at the MetOffice, Hydrol. Earth Syst. Sci., 15, 2729–2746.
 
- Dorigo, W.A., A. Xaver, M. Vreugdenhill, A. Gruber, A. Hegyiova, A.D. Sanchis-Dufau, D. Zamojski, C. Cordes, W. Wagner, and M. Drush, 2012: Global Automated Quality Control of In Situ Soil Moisture Data from the International Soil Moisture Network, Vadoese Zone Journal, doi:10.2136/vzj2012.0097.

- Draper, C. S., Reichle, R. H., De Lannoy, G. J. M., and Liu, Q. (2012), Assimilation of passive and active microwave soil moisture retrievals, Geophys. Res. Lett., 39, L04401, doi:10.1029/2011GL050655.

- Seo, E., M.-I Lee, and R.H. Reichle, 2021: Assimilation of SMAP and ASCAT soil moisture retrievals into the JULES land surface model using the Local Ensemble Transform Kalman Filter, Remote Sensing of Environment, 253, https://doi.org/10.1016/j.rse.2020.112222.

## Requirements

These new Python codes need to be developed. Besides the Python NETCDF4 module, both pygrib and h5py modules are needed for grib1/grib2 and hp5 file opening/reading.  

## Acceptance Criteria (Definition of Done)

All codes were tested within ioda-converters framework and ioda-bundle framework, and their ctest passed. Finally they are merged into the development branch.

## Dependencies

What must be done before this can be done?
 - add issue dependencies in ZenHub as appropriate

Does this block progress on other issues?
 - add this issue as a dependency to other ZenHub issues as appropriate

**Comment by gibbsp** [comment]
@amfox37 I removed this form the ioda-converters Epic, let me know if you are still tracking in AOP20 or now in AOP21.

**Comment by symoore90** [comment]
Closing because epic is completed
